include/aspect/adiabatic_conditions/function.h

```diff
@@ -0,0 +1,107 @@
+/*
+  Copyright (C) 2011, 2012 by the authors of the ASPECT code.
+
+  This file is part of ASPECT.
+
+  ASPECT is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2, or (at your option)
+  any later version.
+
+  ASPECT is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with ASPECT; see the file doc/COPYING.  If not see
+  <http://www.gnu.org/licenses/>.
+*/
+
+
+#ifndef __aspect__adiabatic_conditions_function_h
+#define __aspect__adiabatic_conditions_function_h
+
+
+#include <aspect/adiabatic_conditions/interface.h>
+#include <deal.II/base/point.h>
+#include <deal.II/base/parsed_function.h>
+
+
+namespace aspect
+{
+  namespace AdiabaticConditions
+  {
+    using namespace dealii;
+
+    /**
+     * A simple class that sets the adiabatic conditions based on given
+     * functions.
+     */
+    template <int dim>
+    class Function : public Interface<dim>
+    {
+      public:
+        /**
+         * Constructor. Initialize variables.
+         */
+        Function ();
+
+        /**
+         * Initialization function. Because this function is called after
+         * initializing the SimulatorAccess, all of the necessary information
+         * is available to calculate the adiabatic profile. It computes the
+         * adiabatic conditions along a vertical transect of the geometry
```

Comment:
I think you need to update this string. The material model plays no role for this particular plugin.